nixos/modules/services/networking/anubis.nix

```
@@ -0,0 +1,314 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib) types;
+  jsonFormat = pkgs.formats.json { };
+
+  cfg = config.services.anubis;
+  enabledInstances = (lib.filterAttrs (_: conf: conf.enable) cfg.instances);
+
+  commonSubmodule =
+    isDefault:
+    let
+      mkDefaultOption =
+        path: opts:
+        lib.mkOption (
+          opts
+          // lib.optionalAttrs (!isDefault && opts ? default) {
+            default =
+              lib.attrByPath (lib.splitString "." path)
+                (throw "This is a bug in the Anubis module. Please report this as an issue.")
+                cfg.defaultOptions;
+            defaultText = lib.literalExpression "config.services.anubis.defaultOptions.${path}";
+          }
+        );
+    in
+    { name, ... }:
+    {
+      options = {
+        enable = lib.mkEnableOption "enable this instance of Anubis" // {
+          default = true;
+        };
+        user = mkDefaultOption "user" {
+          default = "anubis";
+          description = "The user under which Anubis is run.";
+          type = types.nullOr types.str;
+        };
+        group = mkDefaultOption "group" {
+          default = "anubis";
+          description = "The group under which Anubis is run.";
+          type = types.nullOr types.str;
+        };
+        dynamicUser = mkDefaultOption "dynamicUser" {
+          default = false;
+          description = "Whether to run Anubis with DynamicUser.";
+          example = true;
+          type = types.bool;
+        };
+
+        botPolicy = lib.mkOption {
+          default = null;
+          description = ''
+            Anubis policy configuration in Nix syntax. Set to `null` to use the baked-in policy which should be
+            sufficient for most use-cases.
+
+            Has no effect if `settings.POLICY_FNAME` is set to a different value, useful for importing an existing
+            configuration.
+
+            See [the documentation](https://anubis.techaro.lol/docs/admin/policies) for details.
+          '';
+          type = types.nullOr jsonFormat.type;
+        };
+
+        extraFlags = mkDefaultOption "extraFlags" {
+          default = [ ];
+          description = "A list of extra flags to be passed to Anubis.";
+          example = [ "-metrics-bind ''" ];
+          type = types.listOf types.str;
+        };
+
+        settings = lib.mkOption {
+          default = { };
+          description = ''
+            Freeform configuration via environment variables for Anubis.
+
+            See [the documentation](https://anubis.techaro.lol/docs/admin/installation) for a complete list of
+            available environment variables.
+          '';
+          type = types.submodule ([
+            {
+              freeformType = types.attrsOf (types.anything);
```

Since we are configuring env vars it may make sense to limit them to this list. Since they should be relatively simple in terms of data. 
```suggestion
              freeformType = types.attrsOf (
                types.nullOr (
                  types.oneOf [
                    types.str
                    types.int
                    types.bool
                  ]
                )
              );
```